force-app/main/default/classes/Data Recipes/SOQLRecipes.cls

```diff
@@ -46,6 +46,29 @@ public inherited sharing class SOQLRecipes {
         return count;
     }
 
+    /**
+     * @description One of the little known features of SOQL for loops is that
+     * you can iterate not only over each record returned by the query, but also
+     * over each *chunk of records*. As the code below demonstrates, specifying
+     * the iteration variable as a list/array will return 200 record chunks
+     * from the query, rather than individual records.
+     * @return      `Integer`
+     */
+    public static Integer[] getChunksOfLargeNumbersOfRecords() {
+        Integer countOfChunks = 0;
+        Integer recordCount = 0;
+
+        for (List<Account> accounts : [
+            SELECT Name
+            FROM Account
+            WITH SECURITY_ENFORCED
+        ]) {
+            countOfChunks += 1;
+            recordCount += accounts.size();
+        }
```

Comment:
I think that this is a bad idea: you're loading unused data from the DB just for sake of counting records and chunks. There's a `COUNT()` function that's way more efficient at that.

You can do the same thing in a much more efficient way with:
```
public static Integer[] getChunksOfLargeNumbersOfRecords() {
  Integer recordCount = [SELECT COUNT() FROM Account WITH SECURITY_ENFORCED];
  Decimal countOfChunks = recordCount / 200;
  return new List<Integer>{
    countOfChunks.round(System.RoundingMode.CEILING).intValue(),
    recordCount
  };
}
```

Reply:
100% agree. However, I'm trying to illustrate how to iterate over the chunks and if i switch to a count, it kind of defeats the purpose.

Reply:
Oh ok, I didn't get that (also partly because of the test code). You may want to add a comment in there to be more explicit about it.